### ISSUE TITLE
labels_target = target_tuple[1].cuda()

### ISSUE THREAD

**Issue by jinx-USTC** [KovenYu/MAR#26]
why use the target label?? is this paper a unsupervised learning manner?

**Comment by KovenYu** [comment]
it is not used in training. just a line of legacy code for some intermediate stats.

**Comment by jinx-USTC** [comment]
Sorry, I don't understand and I miss something. But in your code:

loss_target = self.mdl_loss(features_target, self.multilabel_memory[idx_target], labels_target) ???

Here, in the training, your mdl_loss use the labels_target???

**Comment by KovenYu** [comment]
it is for some stats during developing. you may delete it without changing anything.

**Comment by xiekun2019** [comment]
loss_target = self.mdl_loss(features_target, self.multilabel_memory[idx_target], **labels_target**)
调用的是：
def forward(self, features, multilabels, **labels**):
        P, N = self._partition_sets(features.detach(), multilabels, **labels**)

_partition_sets 的定义：
def _partition_sets(self, features, multilabels, **labels**):
        self._update_buffers(P, **labels**)

_update_buffers的定义：
def _update_buffers(self, P, **labels**):
        if P is None:
            self.n_pos_pairs = 0.9 * self.n_pos_pairs
            return 0
        n_pos_pairs = len(P[0])
        count = 0
        for (i, j) in zip(P[0], P[1]):
            count += **labels**[i] == **labels**[j]
         # labels是用来计算rate_TP，即true positive，并没有用来训练
        **rate_TP** = float(count) / n_pos_pairs
        self.n_pos_pairs = 0.9 * self.n_pos_pairs + 0.1 * n_pos_pairs
        self.rate_TP = 0.9 * self.rate_TP + 0.1 * rate_TP
